cirq-google/cirq_google/calibration/phased_fsim.py

```diff
@@ -1038,3 +1046,49 @@ def try_convert_sqrt_iswap_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedF
         return PhaseCalibratedFSimGate(cirq.FSimGate(theta=np.pi / 4, phi=0.0), 0.5)
 
     return None
+
+
+def try_convert_gate_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedFSimGate]:
+    """Converts a gate to equivalent PhaseCalibratedFSimGate if possible.
+
+    Args:
+        gate: Gate to convert.
+
+    Returns:
+        If provided gate is equivalent to some PhaseCalibratedFSimGate, returns that gate.
+        Otherwise returns None.
+    """
+    phi = 0.0
+    theta = 0.0
+    phase_exponent = 0.0
+    if isinstance(gate, SycamoreGate):
+        phi = np.pi / 6
+        theta = np.pi / 2
+    elif isinstance(gate, cirq.FSimGate):
+        theta = gate.theta
+        phi = gate.phi
+    elif isinstance(gate, cirq.ISwapPowGate):
+        theta = -gate.exponent * np.pi / 2
+    elif isinstance(gate, cirq.PhasedFSimGate):
+        if not np.isclose(gate.zeta, 0.0) or not np.isclose(gate.gamma, 0.0):
+            return None
+        theta = gate.theta
+        phi = gate.phi
+        phase_exponent = -gate.chi / (2 * np.pi)
+    elif isinstance(gate, cirq.PhasedISwapPowGate):
+        theta = -gate.exponent * np.pi / 2
+        phase_exponent = -gate.phase_exponent
+    elif isinstance(gate, cirq.ops.CZPowGate):
+        if not np.isclose(gate.global_shift % (2 * np.pi), 0.0):
```

Comment:
I just realized this property is shared among all gates. Should it be a check shared for all the gates. Does it actually matter at all, why CZ is special?

Reply:
global_shift is not shared among all gates, it's property of the EigenGate: https://quantumai.google/reference/python/cirq/ops/EigenGate
In this method EigenGates are: ISwapPowGate and CZPowGate.

I just did the math and turned out that specifying non-zero global_shift (for any EigenGate) is equivalent to multiplying  gate's matrix by a scalar exp(pi*i*global_shift*exponent).

It does matter, because in case when exp(pi*i*global_shift*exponent) != 1, the input gate is not PhasedFSimGate.

So this is what I will do:
1. Change check to np.isclose(np.exp(np.pi*1j*gate.global_shift*gate.exponent), 1.0);
2. Add a test for non-trivial case (e.g. global_shift=1/3, exponent=3/2); I doubt this will happen on practice, but we will be able to clain that this method translates all gates which can be translated;
3. Do the same check for ISwapPowGate, but not for other gates.

Reply:
Thanks for diving into that!

Reply:
This is now fixed.